pkg/serving/v1alpha1/client.go

```diff
@@ -0,0 +1,160 @@
+// Copyright © 2021 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	knerrors "knative.dev/client/pkg/errors"
+	"knative.dev/client/pkg/util"
+	servingv1alpha1 "knative.dev/serving/pkg/apis/serving/v1alpha1"
+	"knative.dev/serving/pkg/client/clientset/versioned/scheme"
+	clientv1alpha1 "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1alpha1"
+)
+
+// KnServingClient to work with Serving v1alpha1 resources
+type KnServingClient interface {
+	// Namespace in which this client is operating for
+	Namespace() string
+
+	// GetDomainMapping
+	GetDomainMapping(name string) (*servingv1alpha1.DomainMapping, error)
+
+	// CreateDomainMapping
+	CreateDomainMapping(domainMapping *servingv1alpha1.DomainMapping) error
+
+	// UpdateDomainMapping
+	UpdateDomainMapping(domainMapping *servingv1alpha1.DomainMapping) error
+
+	// DeleteDomainMapping
+	DeleteDomainMapping(name string) error
+
+	// ListDomainMappings
+	ListDomainMappings() (*servingv1alpha1.DomainMappingList, error)
+}
+
+type knServingClient struct {
+	client    clientv1alpha1.ServingV1alpha1Interface
+	namespace string
+}
+
+// NewKnServingClient create a new client facade for the provided namespace
+func NewKnServingClient(client clientv1alpha1.ServingV1alpha1Interface, namespace string) KnServingClient {
+	return &knServingClient{
+		client:    client,
+		namespace: namespace,
+	}
+}
+
+func (cl *knServingClient) Namespace() string {
```

Comment:
Add comment doc for all public functions...

Reply:
Done in https://github.com/knative/client/pull/1267/commits/928e2cfb657dbb8011019125942a2270fc8614e7.